src/password-field.tsx

```diff
@@ -69,23 +74,29 @@ const PasswordField: React.FC<PasswordFieldProps> = ({
     ...rest
 }) => {
     const [isVisible, setIsVisible] = React.useState(false);
+    const caretPositionRef = React.useRef<number>();
     const inputRef = React.useRef<HTMLInputElement>(null);
+    const input = inputRef.current;
+    if (input?.selectionStart) {
+        caretPositionRef.current = input.selectionStart;
+    }
```

Comment:
mutating a ref inside render is not safe. It may work today, but it probably won't work in future React versions (with concurrent mode on)